glide-core/redis-rs/redis/src/commands/cluster_scan.rs

```diff
@@ -217,10 +240,20 @@ impl ScanState {
     /// and the address set to the address associated with slot 0.
     /// The address epoch is set to the epoch of the address.
     /// If the address epoch cannot be retrieved, the method returns an error.
-    async fn initiate_scan<C: ClusterInScan + ?Sized>(connection: &C) -> RedisResult<ScanState> {
-        let new_scanned_slots_map: SlotsBitsArray = [0; BITS_ARRAY_SIZE];
+    async fn initiate_scan<C: ClusterInScan + ?Sized>(
+        connection: &C,
+        allow_non_covered_slots: bool,
+    ) -> RedisResult<ScanState> {
+        let mut new_scanned_slots_map: SlotsBitsArray = [0; BITS_ARRAY_SIZE];
         let new_cursor = 0;
-        let address = connection.get_address_by_slot(0).await?;
+        let address = connection
+            .get_address_by_slot(0, &mut new_scanned_slots_map, allow_non_covered_slots)
+            .await?;
+
+        if *address == NO_ADDRESSES_FOUND {
```

Comment:
instead of using NO_ADDRESSES_FOUND, use some enum to hold the result options - for example
```
enum NextNodeResult {
       Address(Arc<String>>),
       AllSlotsCompleted
}